apluslms_roman/configuration.py

```diff
@@ -56,36 +60,11 @@ def find_from(cls, path, name=None, prefixes=None):
                            source=LEGACY_SOURCE)
             raise CourseConfigError("Couldn't find course configuration from {}. Expected to find one of these: {}".format(path, ', '.join(files)))
 
-        return cls.load_from(config)
-
-    @classmethod
-    def load_from(cls, filepath):
-        if not isfile(filepath):
-            raise CourseConfigError("Configuration file {} is not a file".format(filepath))
-
-        with open(filepath) as f:
-            config = load_yaml(f)
-
-        return cls(config, path=dirname(filepath), source=filepath)
-
-    # defaults:
-    steps = ()
-
-    def __init__(self, config, path, source=None, name=None):
-        self.__path__ = path
-        self.__source__ = source
-        self.__config__ = config
-        self.name = name or basename(path)
-
-
-        if not isinstance(config, dict):
-            raise CourseConfigError("Configuration from {} is invalid".format(source or '-'))
-
-        for key, value in config.items():
-            if key[0] == '_':
-                raise CourseConfigError('Invalid option {} in configuration from {}'.format(key, source or '-'))
-            setattr(self, key, value)
+        return super().load(config, version=cls.version, allow_missing=False)
 
+    def steps(self):
+        return self._container._documents[0][1]["steps"]
```

Comment:
Actually, this could probably be a function, though maybe renamed to `get_steps`. Later, filters can be added to it (upcoming feature).

So, later we could use it like `steps = config.get_steps([0, 2, 4, "rst"])` for example.

Reply:
Hmm. There still could be a property `steps` that returns all steps. So,

``´
@property
def steps(self):
  return self.get_steps()
```